pypi-gowin/setup.py

```diff
@@ -0,0 +1,81 @@
+import os
+import subprocess
+import sys
+from setuptools import setup, find_packages
+from setuptools_scm.git import parse as parse_git
+
+def apycula_version():
+    try:
+        try:
+            from importlib import metadata as importlib_metadata  # py3.8+ stdlib
+        except ImportError:
+            import importlib_metadata # py3.7- shim
+    except ImportError:
+        return "" # dummy version for 3.7
```

Comment:
Remove the outer try..except block so it fails completely on 3.6 and lower; we _never_ want the version constraint to not be emitted.

Reply:
Well but how will that work on 3.7? Because it will not successfully execute the setup.py and then not parse the setup_requires.

Reply:
That's what the `import importlib_metadata` shim is for.

Reply:
Now, it won't work at all on 3.6, and that's just fine, since we don't upload sdists. Anyone building YoWASP packages will have to use 3.7 or newer.

Reply:
Yea my point is... how does the shim get installed? It first has to execute setup.py to find out it needs to install that, and if it throws an import error because it's not installed, it wont work. Or I'm misunderstanding how Python resolves the chicken-egg problem of running setup.py to determine the dependencies needed to run setup.py.

Reply:
Oh! It's using an awful workaround for that. Let me check if it actually works on 3.7.

Reply:
Let's just get rid of support for 3.7 and support only Python 3.8+ for building YoWASP.

Reply:
Alright